### PR TITLE
Add common pitfalls section and example in readme

### PR DESCRIPTION
**Pull request by simeg**
I was asked to document this by the very helpful @stevepeak on Gitter. I
am not sure if this is the best format or placement though.

### CODE REVIEW

**Comment by simeg** [comment]
One of the build jobs is failing when uploading reports and is logging this:

`Failed to authenticate repository. Contact staff for more information. [validation-error-t3]`

I'm unable to re-trigger the build.

**Review by eddiemoore — APPROVED** [review]
Just need to fix merge conflicts

**Comment by simeg** [comment]
@eddiemoore Fixed.

File: README.md [hunk 1]
```diff
@@ -161,6 +161,12 @@ There have been reports of [gotwarlost/istanbul](https://github.com/gotwarlost/i
 ## Caveats
 #### Private Repo
 Repository tokens are required for (a) all private repos, (b) public repos not using Travis-CI, CircleCI or AppVeyor. Find your repository token at Codecov and provide via `codecov --token=:token` or `export CODECOV_TOKEN=":token"`
+
+## Common Pitfalls
+
+### [`mock-fs`](https://github.com/tschaub/mock-fs)
+When using `mock-fs` make sure to run `mock.restore()` when your tests are done running, or else the reports wont get generated on Travis CI.
```
Comment (eddiemoore) [inline README.md]: Maybe instead of saying "Travis CI", we say "the CI". As not everyone uses Travis.

Comment (simeg) [reply README.md]: ✅

**Review by eddiemoore — APPROVED** [review]

**Pull request merged**